### PR TITLE
En 3877/bug fix restore blocks into pool set processed mbs

### PR DESCRIPTION
**Pull request by sasurobert**
When restoring a metablock from the storage set all the miniblocks to being processed, as we save into the storage only when the metablock is fully processed.

### CODE REVIEW

File: process/block/shardblock_test.go [hunk 1]
```diff
@@ -4842,3 +4842,84 @@ func TestShardProcessor_GetHighestHdrForOwnShardFromMetachaiMetaHdrsWithOwnHdrSt
 	assert.NotNil(t, hdr)
 	assert.Equal(t, ownHdr.GetNonce(), hdr.GetNonce())
 }
+
+func TestShardProcessor_restoreMetaBlockIntoPoolVerifyMiniblocks(t *testing.T) {
```
Comment (BeniaminDrasovean) [inline process/block/shardblock_test.go]: Could change restore to Restore.

File: process/block/shardblock.go [hunk 2]
```diff
@@ -484,6 +484,11 @@ func (sp *shardProcessor) restoreMetaBlockIntoPool(miniBlockHashes map[string]ui
 			continue
 		}
 
+		processedMiniBlocks := metaBlock.GetMiniBlockHeadersWithDst(sp.shardCoordinator.SelfId())
```
Comment (raduchis) [inline process/block/shardblock.go]: Should this happen also on the metaBlock RestoreFromPool?

File: process/block/shardblock_test.go (hunk 1 above)
Comment (sasurobert) [reply process/block/shardblock_test.go]: done

File: process/block/shardblock.go (hunk 2 above)
Comment (sasurobert) [reply process/block/shardblock.go]: No. Metachain notarizes and processes headers. He does need to look into miniblocks.

Comment (raduchis) [reply process/block/shardblock.go]: ok

Comment (SebastianMarian) [inline process/block/shardblock.go]: I would use the same variable name as it is used in line 526: crossMiniBlockHashes as actually this is why it represents

**Pull request merged**